pkg/deploy/controller/deployment/controller.go

```diff
@@ -144,13 +144,18 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Generate a deployer pod spec.
 		deployerPod, err := c.makeDeployerPod(deployment)
 		if err != nil {
-			return fatalError(fmt.Sprintf("couldn't make deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
+			nextStatus = deployapi.DeploymentStatusFailed
+			updatedAnnotations[deployapi.DeploymentStatusReasonAnnotation] = err.Error()
+			c.emitDeploymentEvent(deployment, kapi.EventTypeWarning, "FailedDecoding", err.Error())
+			break
 		}
 		// Create the deployer pod.
 		deploymentPod, err := c.pn.Pods(deployment.Namespace).Create(deployerPod)
-		// Retry on error.
 		if err != nil {
-			return actionableError(fmt.Sprintf("couldn't create deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
```

Comment:
A create error could be transient if using GenerateName.

On Fri, Sep 2, 2016 at 8:59 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/deploy/controller/deployment/controller.go
> <https://github.com/openshift/origin/pull/10781#discussion_r77341537>:
>
> >  		if err != nil {
> > -			return actionableError(fmt.Sprintf("couldn't create deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
>
> before this change, did the retry re-fetch the deploymentconfig as part of
> the retry (to make sure a deleted namespace or deploymentconfig doesn't
> eternally retry based on internal controller state)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10781/files/967f364efbd38ad9b19131b6ea0b1f26e9374e7c#r77341537>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p-X94VQvf9E5-ijMDu2u61o7mSt6ks5qmB3DgaJpZM4JznXI>
> .
>